Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php

```diff
@@ -397,6 +404,62 @@ public function queryFilterMultiple($data, $clauseType = 'must') {
 	}
 
 	/**
+	 * Adds a simple aggregation with only filed set as configuration
+	 *
+	 * @param $name
+	 * @param $field
+	 * @param string $type
+	 * @param null $parentPath
+	 * @return $this
+	 */
+	public function addSimpleAggregation($name, $field, $type = 'terms', $parentPath = NULL) {
+		if(!array_key_exists("aggregations", $this->request)) {
+			$this->request['aggregations'] = array();
+		}
+
+		$aggregation = array (
+			$type => array (
+				'field' => $field
+			)
+		);
+
+		if($parentPath !== NULL) {
+			$this->addSubAggregation($parentPath, $name, $aggregation);
+		} else {
+			$this->request['aggregations'][$name] = $aggregation;
+		}
+
+		return $this;
+	}
+
+	/**
+	 * This is an low level method for internal usage.
+	 * You can add a custom $aggregationConfiguration under a given $parentPath. The $parentPath foo.bar would
+	 * insert your $aggregationConfiguration under
+	 * $this->request['aggregations']['foo']['aggregations']['bar']['aggregations'][$name]
+	 *
+	 * @param $parentPath
+	 * @param $name
+	 * @param array $aggregationConfiguration
+	 * @return $this
+	 * @throws QueryBuildingException
+	 */
+	public function addSubAggregation($parentPath, $name, $aggregationConfiguration) {
+		// Find the parentPath
+		$path =& $this->request['aggregations'];
+
+		foreach(explode(".", $parentPath) as $subPart) {
```

Comment:
`ObjectAccess::setProperty` and `ArrayHelper` won't work because we can't nest the way we need it. Or do you see a way to use that methods?